File: src/main/java/io/ando5360/app/entity/UserPost.java
package io.ando5360.app.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.io.Serializable;


@Getter
@Setter
@Entity
@Table(name = "posts")
public class UserPost implements Serializable{
	private static final long serialVersionUID = 1L;
	
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long postId;

    @Column(name = "TITLE")
    private String title;

    @Column(name = "CONTENT")
    private String content;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "USER_ID")
    private User user;

}

File: src/main/ng-front/src/app/common/user-post.ts
export class UserPost {
    title: string;
    content: string;
    author: string;
    entityId: string;
    postId : Number;
    userId: Number;

    constructor(title: string, content: string, author: string, entityId: string, postId: Number, userId: Number) {
        this.title = title;
        this.content = content;
        this.author = author;
        this.entityId = entityId;
        this.postId = postId;
        this.userId = userId;
    }
}
